File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
int count_letters(string x);
int count_words(string x);
int count_sentances(string x);
int main(void)
{
    string text = get_string("Text: ");
    int L = round(count_letters(text)/count_words(text) * 100);
    int S = round(count_sentances(text)/count_words(text) * 100);
    int index = 0.0588 * L - 0.296 * S - 15.8;
    if(index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if(index<1)
    {
        printf("Before Grade 1\n");
    }
    else printf("Grade %i\n",index);
}

int count_letters(string x)
{
    int count = 0;
    for(int i = 0; i<strlen(x); i++)
    {
        if(isalpha(x[i])!=0)
        {
            count++;
        }
    }
    return count;
}

int count_words(string x)
{
    float count = 1;
    for(int i = 0; i<strlen(x); i++)
    {
        if(isspace(x[i])!=0)
        {
            count++;
        }
    }
    return count;
}

int count_sentances(string x)
{
    float count = 0;
    for(int i = 0; i<strlen(x); i++)
    {
        if(ispunct(x[i])!=0)
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text: ")


def letters(text):
    count = 0
    for i in range(len(text)):
        if text[i].isalpha() == True:
            count = count + 1
    return float(count)


def words(text):
    count = 1
    for i in range(len(text)):
        if text[i] == ' ':
            count = count + 1
    return float(count)


def sentances(text):
    count = 0
    for i in range(len(text)):
        if text[i] in ['.', '!', '?']:
            count = count + 1
    return float(count)


L = letters(text)/words(text) * 100
S = sentances(text)/words(text) * 100

index = 0.0588 * L - 0.296 * S - 15.8
index = round(index)

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print("Grade", index)
